test/dataloader2/test_proto_multi_rs.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import unittest
+from unittest import TestCase
+
+from torch.utils.data.datapipes.iter.grouping import SHARDING_PRIORITIES
+from torchdata.dataloader2 import DataLoader2, DataLoader2Iterator, PrototypeMultiProcessingReadingService
+
+from torchdata.datapipes.iter import IterableWrapper
+
+
+def _add_one(x: int) -> int:
+    return x + 1
+
+
+class TestPrototypeMultiProcessingReadingService(TestCase):
+    r"""
+    This tests specific functionalities of PrototypeMultiProcessingReadingService, notably
+    `pause`, `resume`, `snapshot`.
+    """
+
+    # Test DataPipes
+    n_elements = 10
+    dp1 = IterableWrapper(range(n_elements)).shuffle().sharding_filter()
+    double_pause_dp = dp1.prefetch().prefetch()
+    test_dps = [dp1, double_pause_dp]
+    mp_start_methods = ["fork", "forkserver", "spawn"]
+
+    def test_reading_service_pause_resume_0_worker(self) -> None:
```

Comment:
nit: can you combine the following two tests by parametrizing the number of workers and prefetch cnts

Reply:
You mean `test_reading_service_pause_resume_0_worker` and `test_reading_service_pause_resume`? I think the expected behaviors are a bit different, hence I separated them

Reply:
I think I can use parameterize to replace the for loop though. I will do that